worldedit-core/src/main/java/com/sk89q/worldedit/extent/world/FastModeExtent.java

```diff
@@ -83,16 +87,36 @@ public void setEnabled(boolean enabled) {
         this.enabled = enabled;
     }
 
+    public boolean isPostEditSimulationEnabled() {
+        return postEditSimulation;
+    }
+
+    public void setPostEditSimulationEnabled(boolean enabled) {
+        this.postEditSimulation = enabled;
+    }
+
     @Override
     public boolean setBlock(BlockVector3 location, BlockStateHolder block) throws WorldEditException {
-        if (enabled) {
+        if (enabled || postEditSimulation) {
             dirtyChunks.add(BlockVector2.at(location.getBlockX() >> 4, location.getBlockZ() >> 4));
-            return world.setBlock(location, block, false);
+
+            if (world.setBlock(location, block, false)) {
+                if (!enabled && postEditSimulation) {
+                    positions.add(location);
+                }
+                return true;
+            }
+
+            return false;
         } else {
             return world.setBlock(location, block, true);
         }
     }
 
+    public boolean commitRequired() {
+        return !dirtyChunks.isEmpty() || !positions.isEmpty();
```

Comment:
Don't forget to include `enabled` in this check, like the others -- just in case.